tests/std/tests/GH_000519_cmath_int_overloads/test.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cmath>
+#include <math.h>
+#include <type_traits>
+
+// Also test the partial implemention of P0533R9 "constexpr for cmath and cstdlib"
+#if _HAS_CXX23
+#define CONSTEXPR23 constexpr
+#else // ^^^ _HAS_CXX23 / !_HAS_CXX23 vvv
+#define CONSTEXPR23 inline
+#endif // ^^^ !_HAS_CXX23 ^^^
+
+CONSTEXPR23 void test_bool_overloads() {
+    // test overloads in std
+
+    assert(std::fpclassify(false) == FP_ZERO);
+    assert(std::fpclassify(true) == FP_NORMAL);
+
+    assert(std::isfinite(false));
+    assert(std::isfinite(true));
+
+    assert(!std::isnan(false));
+    assert(!std::isnan(true));
+
+    assert(!std::isinf(false));
+    assert(!std::isinf(true));
+
+    assert(!std::isnormal(false));
+    assert(std::isnormal(true));
+
+    assert(!std::signbit(false));
+    assert(!std::signbit(true));
+
+    // test overloads in the global namespace
+
+    assert(fpclassify(false) == FP_ZERO);
+    assert(fpclassify(true) == FP_NORMAL);
+
+    assert(isfinite(false));
+    assert(isfinite(true));
+
+    assert(!isnan(false));
+    assert(!isnan(true));
+
+    assert(!isinf(false));
+    assert(!isinf(true));
+
+    assert(!isnormal(false));
+    assert(isnormal(true));
+
+    assert(!signbit(false));
+    assert(!signbit(true));
+}
+
+template <class I>
+CONSTEXPR23 void test_other_integral_overloads() {
+    // test overloads in std
+
+    assert(std::fpclassify(I{}) == FP_ZERO);
+    assert(std::fpclassify(static_cast<I>(42)) == FP_NORMAL);
+    assert(std::fpclassify(static_cast<I>(-42)) == FP_NORMAL);
+
+    assert(std::isfinite(I{}));
+    assert(std::isfinite(static_cast<I>(42)));
+    assert(std::isfinite(static_cast<I>(-42)));
+
+    assert(!std::isnan(I{}));
+    assert(!std::isnan(static_cast<I>(42)));
+    assert(!std::isnan(static_cast<I>(-42)));
+
+    assert(!std::isinf(I{}));
+    assert(!std::isinf(static_cast<I>(42)));
+    assert(!std::isinf(static_cast<I>(-42)));
+
+    assert(!std::isnormal(I{}));
+    assert(std::isnormal(static_cast<I>(42)));
+    assert(std::isnormal(static_cast<I>(-42)));
+
+    assert(!std::signbit(I{}));
+    assert(!std::signbit(static_cast<I>(42)));
+    assert(std::signbit(static_cast<I>(-42)) == std::is_signed_v<I>);
+
+    // test overloads in the global namespace

```

Comment:
Let's also verify that the rest of the functions (`isgreater()` etc.) work with integers, even though the UCRT is doing that for us, and we need to skip the `constexpr` coverage for now.